programs/ziti-edge-tunnel/netif_driver/linux/tun.c

```diff
@@ -47,14 +62,13 @@
 #define IP_BIN "/sbin/ip "
 #endif
 
-#define CHECK_UV(op) do{ int rc = op; if (rc < 0) ZITI_LOG(ERROR, "uv_err: %d/%s", rc, uv_strerror(rc)); } while(0)
+#define _cleanup_(f) __attribute__((cleanup(f)))
```

Comment:
what version of GCC does this require?
is it supported by other compilers (CLang)?

Reply:
I have not found the exact version, however, anecdotal references to GCC having support for "some time" prior to 2008 can be found here: 

http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1298.pdf. 

This is one of the "common" attribute, and I've not been able to produce a gcc invocation that chokes on it (tried -std=c89 -pedantic and a few others):
https://gcc.gnu.org/onlinedocs/gcc/Common-Variable-Attributes.html

Additionally, clang and cci support it. There is a an open issue for MSVC to add support:
https://developercommunity.visualstudio.com/t/add-support-for-gcc-like-cleanup-attribute-for-pla/1187001

Thoughts?